Lib/subprocess.py

```diff
@@ -1307,15 +1307,15 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                 try:
                     exception_name, hex_errno, err_msg = (
                             errpipe_data.split(b':', 2))
+                    err_msg = err_msg.decode(errors="surrogatepass")
                 except ValueError:
                     exception_name = b'SubprocessError'
                     hex_errno = b'0'
```

Comment:
It is safest to assume that anything coming over the errpipe is merely bytes that we cannot interpret in a meaningful manner.  This is an error path, and a rare one at that as all POSIX platforms will be using _posixsubprocess which - being posix async signal safe - is unable to safely do any formatting or processing of error messages.  I would not decode anything.  just leave it as bytes with a repr of the errpipe_data in it.

The details about the error still get across, it doesn't matter if there is an extra b'' in there or not.  feel free to call 

`err_msg = 'Bad exception data from child: ' + repr(bytes(errpipe_data))`

It is a very bad thing to have an exception path that was going to give a useful error message instead fail with a unicode encode/decode error.  I would not call decode at all in tha handler.  The above line is the safe path.  the call to bytes() gets rid of bytearrary() showing up in the repr.  It leaves an ugly "b''" in there - but so what?